concept/type_/relation_type.rs

```diff
@@ -233,7 +239,28 @@ impl<'a> RelationType<'a> {
         type_manager: &TypeManager<Snapshot>,
         name: &str,
     ) -> Result<Option<Relates<'static>>, ConceptReadError> {
-        Ok(self.get_role(snapshot, type_manager, name)?.map(|role_type| Relates::new(self.clone().into_owned(), role_type)))
+        Ok(self
+            .get_role(snapshot, type_manager, name)?
+            .map(|role_type| Relates::new(self.clone().into_owned(), role_type)))
+    }
+
+    pub fn get_relates_transitive<'m, Snapshot: ReadableSnapshot>(
+        &self,
+        snapshot: &Snapshot,
+        type_manager: &'m TypeManager<Snapshot>,
+    ) -> Result<MaybeOwns<'m, HashMap<String, Relates<'static>>>, ConceptReadError> {
+        // TODO: krishnan: Review usage of String as key. It looks strange but fits.
+        type_manager.get_relation_type_relates_transitive(snapshot, self.clone().into_owned())
+    }
+
+    // TODO: Review. This is a bit convoluted thanks to wanting to return the relates with the declaring relation
+    pub fn get_relates_role_transitive<Snapshot: ReadableSnapshot>(
+        &self,
+        snapshot: &Snapshot,
+        type_manager: &TypeManager<Snapshot>,
+        name: &str,
+    ) -> Result<Option<Relates<'static>>, ConceptReadError> {
+        Ok(self.get_relates_transitive(snapshot, type_manager)?.get(name).map(|relates| relates.clone()))
```

Comment:
Needs a review please. It's not as convoluted anymore but still a little strange thanks.